modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/messaging/test/InterestTermsCheckerTest.java

```diff
@@ -0,0 +1,168 @@
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_user = TestPropsValues.getUser();
+
+		Bundle bundle = FrameworkUtil.getBundle(InterestTermsCheckerTest.class);
+
+		_serviceTracker = new ServiceTracker<>(
+			bundle.getBundleContext(),
+			FrameworkUtil.createFilter(
+				"(component.name=com.liferay.segments.asah.connector." +
+					"internal.messaging.InterestTermsChecker)"),
+			null);
+
+		_serviceTracker.open();
+
+		_interestTermsChecker = _serviceTracker.getService();
+
+		Assert.assertNotNull(_interestTermsChecker);
+	}
+
+	@After
+	public void tearDown() {
+		if (_serviceTracker != null) {
+			_serviceTracker.close();
+		}
+	}
+
+	@Test
+	public void testCheckIndividualSegments() throws Exception {
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId", "123456789"
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+				_interestTermsChecker, "_asahFaroBackendClient");
+
+			ReflectionTestUtil.setFieldValue(
+				asahFaroBackendClient, "_http",
+				MockHttpUtil.geHttp(
+					HashMapBuilder.
+						<String, UnsafeSupplier<String, Exception>>put(
+							"/api/1.0/interests/terms/" + _user.getUserId(),
+							() -> JSONUtil.put(
+								"_embedded",
+								JSONUtil.put(
+									"interest-topics",
+									JSONUtil.putAll(
+										JSONUtil.put(
+											"terms",
+											JSONUtil.putAll(
+												JSONUtil.put(
+													"keyword", "term1")))))
+							).put(
+								"page",
+								JSONUtil.put(
+									"number", 0
+								).put(
+									"size", 100
+								).put(
+									"totalElements", 1
+								).put(
+									"totalPages", 1
+								)
+							).put(
+								"total", 0
+							).toString()
+						).build()));
+
+			ReflectionTestUtil.invoke(
+				_interestTermsChecker, "checkInterestTerms",
+				new Class<?>[] {long.class, String.class}, _user.getCompanyId(),
+				String.valueOf(_user.getUserId()));
+
+			PortalCache<String, String[]> portalCache =
+				(PortalCache<String, String[]>)_multiVMPool.getPortalCache(
+					"com.liferay.segments.asah.connector.internal.cache." +
+						"AsahInterestTermCache");
+
+			String[] terms = portalCache.get("segments-" + _user.getUserId());
```

Comment:
You don't need to declare terms, as it is only used for the assert

Reply:
Solved. Thanks